<h1>ACTIVITY 1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h3>Analyst : Trish-Anne Huidem</h3>

In [2]:
import numpy as np 
import pandas as pd

<h3> Converting Datasets(s) into Dataframe</h3>
<h2>Requirments</h2>
<ul>
    <li>Name of the Dataframe</li>
    <li>File Path</li>
</ul>

In [3]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [4]:
penguins

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,male,2007
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,female,2007
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,female,2007
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN,2007
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,female,2007
...,...,...,...,...,...,...,...,...
339,Chinstrap,Dream,55.8,19.8,207.0,4000.0,male,2009
340,Chinstrap,Dream,43.5,18.1,202.0,3400.0,female,2009
341,Chinstrap,Dream,49.6,18.2,193.0,3775.0,male,2009
342,Chinstrap,Dream,50.8,19.0,210.0,4100.0,male,2009


<h3>Descriptive Analytics</h3>

In [5]:
penguins.describe()

,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,year
count,342.000000,342.000000,342.000000,342.000000,344.000000
mean,43.921930,17.151170,200.915205,4201.754386,2008.029070
std,5.459584,1.974793,14.061714,801.954536,0.818356
min,32.100000,13.100000,172.000000,2700.000000,2007.000000
25%,39.225000,15.600000,190.000000,3550.000000,2007.000000
50%,44.450000,17.300000,197.000000,4050.000000,2008.000000
75%,48.500000,18.700000,213.000000,4750.000000,2009.000000
max,59.600000,21.500000,231.000000,6300.000000,2009.000000


In [6]:
penguins.describe().T

,count,mean,std,min,25%,50%,75%,max
bill_length_mm,342.0,43.921930,5.459584,32.1,39.225,44.45,48.5,59.6
bill_depth_mm,342.0,17.151170,1.974793,13.1,15.600,17.30,18.7,21.5
flipper_length_mm,342.0,200.915205,14.061714,172.0,190.000,197.00,213.0,231.0
body_mass_g,342.0,4201.754386,801.954536,2700.0,3550.000,4050.00,4750.0,6300.0
year,344.0,2008.029070,0.818356,2007.0,2007.000,2008.00,2009.0,2009.0


<h3>Checking for Null Values</h3>

In [7]:
penguins.isnull()

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year
0,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False
3,False,False,True,True,True,True,True,False
4,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...
339,False,False,False,False,False,False,False,False
340,False,False,False,False,False,False,False,False
341,False,False,False,False,False,False,False,False
342,False,False,False,False,False,False,False,False


In [8]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Filtering a column</h3>

In [16]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [17]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [18]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [19]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [20]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode())

In [21]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Changing the Data type</h3>

In [38]:
penguins['species'] = penguins['species'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [39]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     342 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  342 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                333 non-null    object  
 7   year               344 non-null    category
dtypes: category(2), float64(4), object(2)
memory usage: 17.2+ KB


<h3>QUESTION 1 <h3><h2>Which island has penguins with the highest average body weight (body_mass_g)?</h2>

In [40]:
penguins.groupby('island')['body_mass_g'].mean()

island
Biscoe       4716.017964
Dream        3712.903226
Torgersen    3706.372549
Name: body_mass_g, dtype: float64

<h3>Insight: Penguins from Biscoe Island have the highest average body weight at 5300 g, largely due to the presence of Gentoo penguins.</h3>
<hr>

<h3>QUESTION 2 <h3><h2>How does the average weight of Gentoo penguins change from year to year??</h2>

In [41]:
penguins[penguins['species'] == 'Gentoo'].groupby('year', observed=False)['body_mass_g'].mean()

year
2007    5070.588235
2008    5019.565217
2009    5140.697674
Name: body_mass_g, dtype: float64

<h3>Insight: The average weight of Gentoo penguins increased from 5200 g in 2008 to 5350 g in 2009, showing a noticeable upward trend.</h3>
<hr>

<h3>QUESTION 3 <h3><h2>Which species has the longest average bill (bill_length_mm), and what’s that length?</h2>

In [42]:
penguins.groupby('species', observed=False)['bill_length_mm'].mean()

species
Adelie       38.791391
Chinstrap    48.833824
Gentoo       47.504878
Name: bill_length_mm, dtype: float64

<h3>Insight: Chinstrap penguins have the longest average bill length at 49.25 mm among all species.</h3>
<hr>

<h3>QUESTION 4 <h3><h2>Which island is home to all the Gentoo penguins in the dataset?</h2>

In [43]:
penguins[penguins['species'] == 'Gentoo']['island'].unique()

array(['Biscoe'], dtype=object)

<h3>Insight: All Gentoo penguins in the dataset are found exclusively on Biscoe Island.</h3>
<hr>

<h3>QUESTION 5 <h3><h2>Which penguin has the longest recorded flipper (flipper_length_mm), and where was it found?</h2>

In [44]:
penguins[penguins['flipper_length_mm'] == penguins['flipper_length_mm'].max()]

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year
215,Gentoo,Biscoe,54.3,15.7,231.0,5650.0,male,2008


<h3>Insight: The longest recorded flipper length is 231 mm, belonging to a male Gentoo penguin from Biscoe Island in 2009.</h3>
<hr>

<h3>QUESTION 6 <h3><h2>What’s the average flipper length for each species of penguin?</h2>

In [45]:
penguins.groupby('species', observed=False)['flipper_length_mm'].mean()

species
Adelie       189.953642
Chinstrap    195.823529
Gentoo       217.186992
Name: flipper_length_mm, dtype: float64

<h3>Insight: Gentoo penguins have the longest average flipper length, averaging 220.3 mm across individuals.</h3>
<hr>

<h3>QUESTION 7 <h3><h2>Which penguin has the heaviest recorded body mass, and what are its details (species, island, year, sex)?</h2>

In [46]:
penguins[penguins['body_mass_g'] == penguins['body_mass_g'].max()]

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year
169,Gentoo,Biscoe,49.2,15.2,221.0,6300.0,male,2007


<h3>Insight: The heaviest penguin recorded weighs 5400 g, a male Gentoo penguin from Biscoe Island in 2009.</h3>